src/core/image.js

```diff
@@ -506,6 +506,109 @@ var PDFImage = (function PDFImageClosure() {
       }
     },
 
+    /**
+     * Added by RamSoft (11/5/2015) to resize large resolution PDFS 
+     * as to improve rendering time
+     */
+    resizeGrayPixels: function(imgData, comps, bpc) {
+      var result = false;
+
+      //we only resize 1 bit per pixel or 8 bits per pixel grayscale image
+      if (comps === 1 && (bpc === 1 || bpc === 8)) {
+        var scaleBits;
+        var h1 = imgData.height;
+        var w1 = imgData.width;
+
+        if ((h1 > 15000) || (w1 > 15000)) {
+          scaleBits = 3;
+        } else if ((h1 > 10000) || (w1 > 10000)) {
+          scaleBits = 2;
+        } else if ((h1 > 5000) || (w1 > 5000)) {
+          scaleBits = 1; // 25% - 4
+        } else {
+          scaleBits = 0;
+        }
+
+        if (scaleBits > 0) {
+          var w2 = w1 >> scaleBits;
+          var h2 = h1 >> scaleBits;
+
+          var newRowBytes = (w2 * comps * bpc + 7) >> 3;
+          var originalRowBytes = (w1 * comps * bpc + 7) >> 3;
+
+          var numBytes = h2 * newRowBytes;
+          var pixelArrayOutput = new Uint8Array(numBytes);
+          var x2, y2;
+          var originalRowStart;
+          var newRowStart;
+          var i, j;
+
+          if (bpc === 1) {
+
+            for (i = 0; i < h2; i++) {
+              y2 = i << scaleBits;
```

Comment:
here and below for i/x2 and j/y2 pairs `for (i = 0, y2 = 0; i < h2; i++, y2 += step) {` to remove `i << scaleBits` calculations (`var step = 1 << scaleBits`)